src/cryptonote_core/blockchain.cpp

```diff
@@ -1044,6 +1044,11 @@ bool Blockchain::switch_to_alternative_blockchain(std::list<block_extended_info>
     reorg_notify->notify("%s", std::to_string(split_height).c_str(), "%h", std::to_string(m_db->height()).c_str(),
         "%n", std::to_string(m_db->height() - split_height).c_str(), "%d", std::to_string(discarded_blocks).c_str(), NULL);
 
+  std::shared_ptr<tools::Notify> block_notify = m_block_notify;
```

Comment:
I don't see any benefit to copying the `std::shared_ptr` in the original code. The copy is not atomic/thread-safe (with respect to assignment of the same `std::shared_ptr` object) unless the appropriate functions are used. Was `std::atomic_store` and `std::atomic_load` intended to be used here?

Reply:
Sorry ping @moneromooo-monero

Reply:
Yes, I thought the pointer copy was atomic.

Reply:
I think the risks of issue here is low due to the init process. But something to remember for sure.